### PR TITLE
added updatePreviousAccessLevels when creating new version of report 

### PR DESCRIPTION
**Pull request by StuyckRuben**
added strengthenAccessLevelToInternRegering for the previousVersions when decisionResult is updated to UITGESTELD or INGETROKKEN

https://kanselarij.atlassian.net/browse/KAS-4208

### CODE REVIEW

File: app/components/agenda/agendaitem/agendaitem-decision.js [hunk 1]
```diff
@@ -149,6 +150,13 @@ export default class AgendaitemDecisionComponent extends Component {
           piece
         );
       }
+      let reportToUpdate = this.report;
+      while (reportToUpdate) {
+        await this.pieceAccessLevelService.strengthenAccessLevelToInternRegering(
+          reportToUpdate
+        );
+        reportToUpdate = await reportToUpdate.previousPiece
+      }
```
Comment (sergiofenoll) [inline app/components/agenda/agendaitem/agendaitem-decision.js]: Is the `while` loop necessary? `strengthenAccessLevelToInternRegering` should already update the previous versions.

**Review by sergiofenoll — CHANGES_REQUESTED** [review]

**Review by ValenberghsSven — CHANGES_REQUESTED** [review]
Looking at this I might have made mistake when creating this ticket.
I said that the report needs to be updated when changing the decision-result-code.
However I am not sure this is correct.
Trying to puzzle together why I thought that.
in this ticket it is clearly stated that the access rights on reports stay unchanged
https://kanselarij.atlassian.net/browse/KAS-2008

Also mentioned here that report rights should stay unchanged
https://kanselarij.atlassian.net/browse/KAS-3826

We do need to update when adding BIS
https://kanselarij.atlassian.net/browse/KAS-3938

I might had it confused with setting confidential > access levels of report should become confidential.
`updateDecisionsAccessLevelOfSubcase`
But that feature should already be ok and also updates previous versions. Going to test that

File: app/components/agenda/agendaitem/agendaitem-decision.js (hunk 1 above)
Comment (ValenberghsSven) [reply app/components/agenda/agendaitem/agendaitem-decision.js]: see my comment. I think this entire block should go.
Report access levels stay unchanged (but there should be a difference between the original report and a BIS report)

**Review by ValenberghsSven — APPROVED** [review]
Bit biased, but only 1 line needs to be added.
Tested it and it works.

**Pull request merged**